lib/signaling/v2/networkqualitymonitor.js

```diff
@@ -0,0 +1,119 @@
+/* eslint callback-return:0 */
+'use strict';
+
+const EventEmitter = require('events');
+
+const PeerConnectionReportFactory = require('../../stats/peerconnectionreportfactory');
+
+/**
+ * @emits NetworkQualityMonitor#updated
+ */
+class NetworkQualityMonitor extends EventEmitter {
+  /**
+   * Construct a {@link NetworkQualityMonitor}.
+   * @param {PeerConnectionManager} manager
+   * @param {NetworkQualitySignaling} signaling
+   */
+  constructor(manager, signaling) {
+    super();
+    Object.defineProperties(this, {
+      _factories: {
+        value: new WeakMap()
+      },
+      _manager: {
+        value: manager
+      },
+      _signaling: {
+        value: signaling
+      },
+    });
+    signaling.on('updated', () => this.emit('updated'));
+  }
+
+  /**
+   * Get the current {@link NetworkQualityLevel}, if any.
+   * @returns {?NetworkQualityLevel} level - initially null
+   */
+  get level() {
+    return this._signaling.level;
+  }
+
+  /**
+   * Get the current {@link NetworkQualityLevels}, if any.
+   * @deprecated - the decomposed levels are only used for debugging and will be
+   *   removed as soon as we are confident in our implementation
+   * @returns {?NetworkQualityLevels} levels - initially null
+   */
+  get levels() {
+    return this._signaling.levels;
+  }
+
+  /**
+   * Start monitoring.
+   * @returns {void}
+   */
+  start() {
+    this.stop();
+    const timeout = setTimeout(() => {
+      if (this._timeout !== timeout) {
+        return;
+      }
+      next(this).then(reports => {
+        if (this._timeout !== timeout) {
+          return;
+        }
+        if (reports.length) {
+          const [report] = reports;
+          this._signaling.put(report);
+        }
+        this.start();
+      });
+    }, 200);
+    this._timeout = timeout;
+  }
+
+  /**
+   * Stop monitoring.
+   * @returns {void}
+   */
+  stop() {
+    clearTimeout(this._timeout);
+    this._timeout = null;
+  }
+}
+
+/**
+ * @param {NetworkQualityMonitor}
+ * @returns {Promise<NetworkQualityInputs>}
+ */
+function  next(monitor) {
```

Comment:
@markandrus Nitpick: extra space between `function` and `next`.

Reply:
Oops, thanks